### DISCUSSION TITLE
autocorrect issue with $ character

### DISCUSSION THREAD

**Discussion by unbroken75** [dbeaver/dbeaver#36409] [DBeaver Development]
In my ERP DB all field names start with a t$ prefix and $ is considered a special character by DBeaver, so autocorrect replaces any t$xxx to "T$XXX" with double quotes. 
Having all fields name enclosed in double quote is quite an hassle and useless in my case, because 'select t$xxx from table' works perfectly as 'select "T$XXX" from table'
How can I prevent DBeaver from enclosing every fields in double quotes without disabling autocorrect ?
Thank you,

**Comment by ged-yuko** [comment]
As far as i understand, '$' character cannot be a part of the valid unquotted identifier according to the SQL Standard. If your database supports that, then it is a DB-specific custom extension of the SQL syntax. DBeaver just cannot support all the custom SQL extensions of all the databases it is able to connect, so it uses standard escaping rules while generating and parsing SQL, until a lack of certain syntax support totally prevents some necessary function of database from being usable. 

There is also a special treatment of '$' character in the context of client-side scripting, which afftects parsing logic, so it might not even be possible at all to support these two scenarios of this character usage at the same time.
